Sources/MongoSwift/BSON/Document.swift

```diff
@@ -79,6 +89,9 @@ public struct Document: ExpressibleByDictionaryLiteral, ExpressibleByArrayLitera
         }
 
         self.storage = DocumentStorage()
+        // This is technically not consistent, but the only way this inconsistency can cause an issue is if we fail to
+        // setValue(), in which case we crash anyways.
```

Comment:
Can you elaborate on why this is technically not consistent? Is it because we're initializing it to 0?

I would think that's fine provided that `self.setValue()` increments the count as new keys are assigned. Otherwise, we're correct to initialize to zero if `keyValuePairs` is empty and this `for` loop has nothing to iterate.

Reply:
That's it. It is not consistent at the time of setting it to 0, because we may actually have more key value pairs. `setValue()` fixes this once it is called enough times.

This is okay because if `setValue()` fails, we crash so it doesn't matter.